source/PlayerInfo.cpp

```diff
@@ -74,6 +74,33 @@ namespace {
 			ships[0] = flagship;
 		}
 	}
+
+	string EntryToString(SystemEntry entryType)
+	{
+		switch(entryType)
+		{
+			case SystemEntry::HYPERDRIVE:
+				return "hyperdrive";
+			case SystemEntry::JUMP:
+				return "jump drive";
+			case SystemEntry::WORMHOLE:
+				return "wormhole";
+			default:
+			case SystemEntry::TAKE_OFF:
+				return "takeoff";
+		}
+	}
+
+	SystemEntry StringToEntry(const string &entry)
+	{
+		if(entry == "hyperdrive")
+			return SystemEntry::HYPERDRIVE;
+		else if(entry == "jump drive")
+			return SystemEntry::JUMP;
+		else if(entry == "wormhole")
+			return SystemEntry::WORMHOLE;
+		return SystemEntry::TAKE_OFF;
+	}
```

Comment:
PR https://github.com/endless-sky/endless-sky/pull/7348 will need an update if this PR gets merged first.

Also; the player could also have entered a system because it is a new player and just didn't leave the planet yet at all. (SystemEntry::BORN_HERE_AND_NEVER_LEFT_YET is a bit long though...)